### PR TITLE
feat: add customizable ttl_bump interval

### PR DESCRIPTION
**Pull request by Emyrk**
Closed for https://github.com/coder/coder/pull/10704

If `ttl_bump` is set, allows the activity of a workspace to bump the TTL by some other amount than the defaut TTL.

Example being, if you have autostop set to 8hrs, then the workspace deadline is set to 8hrs.
When the workspace has any activity, the deadline increases by 8hrs. If done at the last minute, this means the deadline is 16hours out from the original amount.

This PR allows setting the autostop to 8hrs, with activity bumping the deadline by a smaller amount, such as 1 hour. 

This is especially useful for long autostops.

**By default** ttl_bump is set to 0 and **does nothing**

# UI

[Peek 2023-11-13 12-09.webm](https://github.com/coder/coder/assets/5446298/2a3c5fb1-3926-485e-830e-e828005e2351)

Closes https://github.com/coder/customers/issues/318

### CODE REVIEW

File: coderd/database/migrations/000169_ttl_bump.up.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BEGIN;
+
+DROP VIEW template_with_users;
+
+ALTER TABLE templates ADD COLUMN default_ttl_bump bigint DEFAULT '0'::bigint NOT NULL;
+COMMENT ON COLUMN templates.default_ttl_bump IS 'Amount of time to bump workspace ttl from activity. 0 will default to the "default_ttl" as the bump interval.';
+
+
+ALTER TABLE workspaces ADD COLUMN ttl_bump bigint DEFAULT '0'::bigint NOT NULL;
+COMMENT ON COLUMN workspaces.ttl_bump IS 'Amount of time to bump workspace ttl from activity. 0 will default to the "ttl" as the bump interval.';
```
Comment (Emyrk) [inline coderd/database/migrations/000169_ttl_bump.up.sql:10]: 0 means defer to `ttl` for the bump interval.

**Comment by bpmct** [comment]
This is awesome

**Comment by Emyrk** [comment]
Actually changing this up in favor of something more simple.

**Comment by Emyrk** [comment]
Closed for https://github.com/coder/coder/pull/10704

**Pull request closed without merging**